### PR TITLE
LibGUI: Allow blocking modals and popups to handle their own shortcuts

### PR DESCRIPTION
**Pull request by LucasChollet**
If you ever wondered why `Ctrl-H` wasn't working in `FilePicker`, here is a fix :smile:.

### CODE REVIEW

File: Userland/Libraries/LibGUI/Window.h [hunk 1]
```diff
@@ -229,7 +229,12 @@ class Window : public Core::Object {
 
     void set_always_on_top(bool always_on_top = true);
 
-    void propagate_shortcuts_up_to_application(KeyEvent& event, Widget* widget);
+    enum class PropagationLimit {
```
Comment (konradekk) [inline Userland/Libraries/LibGUI/Window.h:232]: `EventTargetPropagationBoundary`? (or even without `Boundary`?)

File: Userland/Libraries/LibGUI/Window.cpp [hunk 2]
```diff
@@ -499,7 +499,7 @@ void Window::propagate_shortcuts_up_to_application(KeyEvent& event, Widget* widg
 
     if (!action)
         action = action_for_shortcut(shortcut);
-    if (!action)
+    if (!action && limit == PropagationLimit::Application)
```
Comment (konradekk) [inline Userland/Libraries/LibGUI/Window.cpp:502]: technically this should probably be
```suggestion
    if (!action && limit >= PropagationLimit::Application)
```
and the previous one would be then
```cpp
    if (!action && limit >= PropagationLimit::CurrentWindow)
```

File: Userland/Libraries/LibGUI/Window.h [hunk 3]
```diff
@@ -229,7 +229,12 @@ class Window : public Core::Object {
...
+        CurrentWindow,
+        Application,
```
Comment (konradekk) [inline Userland/Libraries/LibGUI/Window.h:234]: not really symmetric for the `Current` part (my opinion: go for removal 🙃)

File: Userland/Libraries/LibGUI/Window.cpp [hunk 4]
```diff
@@ -479,7 +479,7 @@ void Window::handle_multi_paint_event(MultiPaintEvent& event)
         ConnectionToWindowServer::the().async_did_finish_painting(m_window_id, rects);
 }
 
-void Window::propagate_shortcuts_up_to_application(KeyEvent& event, Widget* widget)
+void Window::propagate_shortcuts(KeyEvent& event, Widget* widget, ShortcutPropagationBoundary limit)
```
Comment (konradekk) [inline Userland/Libraries/LibGUI/Window.cpp:482]: `limit` → `boundary` to be more coherent? 😅

**Review by konradekk — APPROVED** [review]
lgtm.

**Review by GMTA — APPROVED** [review]
Nice!

**Pull request merged**